src/data/repositories/IncidentManagementTeamD2Repository.ts

```diff
@@ -0,0 +1,254 @@
+                        },
+                    })
+                )
+                    .flatMap(response =>
+                        assertOrError(response.users[0], "Incident Management Team Member")
+                    )
+                    .map(d2User =>
+                        this.mapUserToIncidentManagementTeamMember(
+                            d2User as D2UserFix,
+                            events,
+                            dataElementRoles
+                        )
+                    );
+            }
+        );
+    }
+
+    private mapUserToIncidentManagementTeamMember(
+        d2User: D2UserFix,
+        events: D2TrackerEvent[],
+        dataElementRoles: D2DataElement[]
+    ): TeamMember {
+        const avatarId = d2User?.avatar?.id;
+        const photoUrlString = avatarId
+            ? `${this.api.baseUrl}/api/fileResources/${avatarId}/data`
+            : undefined;
+
+        const teamMember = new TeamMember({
+            id: d2User.id,
+            username: d2User.username,
+            name: d2User.name,
+            email: d2User.email,
+            phone: d2User.phoneNumber,
+            status: "Available", // TODO: Get status when defined
+            photo:
+                photoUrlString && TeamMember.isValidPhotoUrl(photoUrlString)
+                    ? new URL(photoUrlString)
+                    : undefined,
+            teamRoles: undefined,
+            workPosition: undefined, // TODO: Get workPosition when defined
+        });
+
+        const teamRoles = getTeamMemberIncidentManagementTeamRoles(
+            teamMember,
+            events,
+            dataElementRoles
+        );
+
+        return new TeamMember({
+            ...teamMember,
+            teamRoles: teamRoles.length > 0 ? teamRoles : undefined,
+        });
+    }
+
+    saveIncidentManagementTeamMemberRole(
+        teamMemberRole: TeamRole,
+        incidentManagementTeamMember: TeamMember,
+        diseaseOutbreakId: Id
+    ): FutureData<void> {
+        return this.saveOrDeleteIncidentManagementTeamMember(
+            teamMemberRole,
+            incidentManagementTeamMember,
+            diseaseOutbreakId,
+            "CREATE_AND_UPDATE"
+        );
+    }
+
+    deleteIncidentManagementTeamMemberRole(
+        teamMemberRole: TeamRole,
+        incidentManagementTeamMember: TeamMember,
+        diseaseOutbreakId: Id
+    ): FutureData<void> {
+        return this.saveOrDeleteIncidentManagementTeamMember(
+            teamMemberRole,
+            incidentManagementTeamMember,
+            diseaseOutbreakId,
+            "DELETE"
+        );
+    }
+
+    private getDataElementRolesAndIncidentManagementTeamEvents(diseaseOutbreakId: Id): FutureData<{
+        dataElementRoles: D2DataElement[];
+        events: D2TrackerEvent[];
+    }> {
+        return Future.joinObj(
+            {
+                dataElementRoles: apiToFuture(
+                    this.api.models.dataElements.get({
```

Comment:
this call should be made in RoleD2Repository as it is related to fetching roles. Two different repositories can be called in the use case and then any mapping between them can be done in use case.